### PR TITLE
feat: Support option historical data

### PR DESCRIPTION
**Pull request by hiohiohio**
Context:
- market data for options are introduced

Changes:
- add OptionHistoricalDataClient

### CODE REVIEW

File: alpaca/data/historical/option.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+            secret_key=secret_key,
+            oauth_token=oauth_token,
+            use_basic_auth=use_basic_auth,
+            api_version="v1beta1",
+            base_url=url_override if url_override is not None else BaseURL.DATA,
+            sandbox=False,
+            raw_data=raw_data,
+        )
+
+    def get_option_quotes(
```
Comment (gnvk) [inline alpaca/data/historical/option.py:75]: This is not implemented yet on the server side.

File: alpaca/data/historical/option.py [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+
+        if self._use_raw_data:
+            return raw_latest_quotes
+
+        return parse_obj_as_symbol_dict(Quote, raw_latest_quotes)
+
+    def get_option_latest_trade(
+        self, request_params: OptionLatestTradeRequest
+    ) -> Union[Dict[str, Trade], RawData]:
+        """Retrieves the latest quote for an option symbol or list of option symbols.
```
Comment (gnvk) [inline alpaca/data/historical/option.py:156]: quote -> trade

File: alpaca/data/historical/option.py [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+            extension=DataExtensionType.SNAPSHOT,
+            **params,
+        )
+
+        if self._use_raw_data:
+            return raw_snapshots
+
+        return parse_obj_as_symbol_dict(Snapshot, raw_snapshots)
+
+    # TODO: Remove duplication
```
Comment (gnvk) [inline alpaca/data/historical/option.py:233]: This should be done!

File: alpaca/data/models/quotes.py [hunk 4]
```diff
@@ -34,7 +34,7 @@ class Quote(BaseModel):
     bid_exchange: Optional[Union[str, Exchange]] = None
     bid_price: float
     bid_size: float
-    conditions: Optional[List[str]] = None
+    conditions: Optional[Union[List[str], str]] = None
```
Comment (gnvk) [inline alpaca/data/models/quotes.py:37]: I don't know how precise we want to be. For example options don't have a tape.

File: tests/data/test_historical_option_data.py [hunk 5]
```diff
@@ -0,0 +1,774 @@
...
+                "ax": "K",
+                "ap": 158.65,
+                "as": 1,
+                "bx": "Q",
+                "bp": 159.52,
+                "bs": 4,
+                "c": [
+                    "R"
+                ],
+                "z": "C"
```
Comment (gnvk) [inline tests/data/test_historical_option_data.py:45]: This is not an option response. Same is true many times in this file.

File: alpaca/data/historical/option.py (hunk 3 above)
Comment (hiohiohio) [reply alpaca/data/historical/option.py:233]: Yeah, please let me refactor this in the followup PR. I have modified a bit for options. Therefore, not exactly matching for stock/crypto/options now.

File: alpaca/data/historical/option.py (hunk 1 above)
Comment (hiohiohio) [reply alpaca/data/historical/option.py]: Thank you! commented out until implemented on the server side.

File: alpaca/data/historical/option.py (hunk 2 above)
Comment (hiohiohio) [reply alpaca/data/historical/option.py]: fixed

File: tests/data/test_historical_option_data.py (hunk 5 above)
Comment (hiohiohio) [reply tests/data/test_historical_option_data.py]: Thank you! fixed responses for unit test cases.

File: tests/data/test_historical_option_data.py [hunk 6]
```diff
@@ -0,0 +1,774 @@
...
+            "c": [
+                "@"
+            ],
+            "i": 22730,
+            "z": "C"
+        }
+    }
+        """,
+    )
+    request = OptionLatestTradeRequest(symbol_or_symbols=symbol, feed=DataFeed.IEX)
```
[the remark this answers is not in the document]
Comment (hiohiohio) [reply tests/data/test_historical_option_data.py]: Thank you! remove them.

**Review by gnvk — APPROVED** [review]
LGTM

**Pull request merged**